docs/contract-test-library.md

```diff
@@ -0,0 +1,541 @@
+---
+title: Taquito Smart Contract Library
+id: contracts_library
+author: Michael Kernaghan
+---
+
+The contracts used in Taquito Integration Tests and in Taquito Documentation Live Code Examples are test data and require curation. Here we collect the contracts, give them names, demonstrate their properties and describe their use.
+
+To determine if a contract has an FA1.2 interface we can use 
+```
+tezos-client check contract KT1CfFBaLoUrgv93k8668KCCcu2hNDNYPz4L implements fa1.2
+```
+A contract has an FA2 interface if it has entrypoints: transfer, balance_of, and update_operators
+- **Basic Contracts**
+  - [IncrementContract](#incrementcontract)
+  - [MichelsonMapContract](#michelsonmapcontract)
+- **Lambda Contracts**
+  - [LambdaViewContract](#lambdaviewcontract)
+  - [LambdaViewWithTokenContract](#lambdaviewwithtokencontract)
+- **Map and BigMap Contracts**
+  - [MapWithPairasMapContract](#mapwithpairasmapcontract)
+  - [MapWithValuesComplexKeysContract](#mapwithvaluescomplexkeyscontract)
+  - [MapWithInitialStorageContract](#mapwithinitialstoragecontract)
+  - [MapWithMapandBigmapContract](#mapwithmapandbigmapcontract)
+  - [BigMapsMultipleValuesContract](#bigmapsmultiplevaluescontract)
+  - [BigMapsComplexStorageContract](#smartcontractcomplexstoragecontract)
+  - [BigMapPackContract](#bigmappackcontract)
+- **Tzip7 Contracts**
+  - [TokenContract](#tokencontract)
+- **Tzip12 Contracts**
+  - [Tzip12BigMapOffChainContract](#tzip12bigmapoffchaincontract)
+- **Tzip 16 Contracts**
+  - [Tzip16StorageContract](#tzip16storagecontract)
+  - [Tzip16HTTPSContract](#tzip16httpscontract)
+  - [Tzip16SHA256Contract](#tzip16sha256contract)
+  - [Tzip16IPFSContract](#tzip16ipfscontract)
+  - [Tzip16OffChainContractOne](#tzip16offchaincontractone)
+  - [Tzip16OffChainContractTwo](#tzip16offchaincontracttwo)
+- **Wallet Contracts**
```

Comment:
Do these contracts are called wallet contracts because they are used on the wallet documentation page?

Reply:
yes